### ISSUE TITLE
`cloudflared tunnel run` stopped working for no apparent reason

### ISSUE THREAD

**Issue by Changaco** [cloudflare/cloudflared#400]
The logs were full of this:

`error="Unable to reach the origin service. The service may be down or it may not be responding to traffic from cloudflared: unsupported protocol scheme \"\"" cfRay=… originService="unix socket: /path/to/socket"`

…but the web app was working. In fact at first the error only occurred for some of the requests, while others were being forwarded successfully. All requests were failing after I restarted cloudflared, and the problem persisted after I restarted the web app and confirmed with `curl --unix-socket /path/to/socket` that it was working.

In the end I was able to make it work again by modifying the web app to listen on `127.0.0.1:8000` instead of the Unix socket, and reconfiguring cloudflared accordingly.

The cloudflared version was 2021.5.8, running with the `--no-autoupdate` argument. I tried updating to 2021.5.10 but it didn't seem to change anything.

I haven't been able to find the cause of this incident, but there is at least one thing that should be fixed in cloudflared: the unhelpful error message `unsupported protocol scheme ""`.

**Comment by eidam** [comment]
I am having the same issue, without any luck debugging what caused it

**Comment by eidam** [comment]
I think asking for help magically helped to figure it out. 

I just changed the cloudflared protocol to `h2mux` and it is working once again (with the latest cloudflared)
Seems like something changed and the default protocol was changed to `http2`, which breaks the unix sockets as the origin.

tldr set `--protocol h2mux` either in the cmd or config file and you should be fine.

**Comment by nmldiegues** [comment]
Thank you for the report. We will look into this soon.

**Comment by eidam** [comment]
Thank you! Let me know if I can help you anyhow.

**Comment by nmldiegues** [comment]
We've released cloudflared 2021.6.0 (https://github.com/cloudflare/cloudflared/commit/31ff7caeebfc7fd9d81bcce4538dedc7ba978221) that fixes this problem and improved our end to end tests to consider unix socket origins.

Thanks for flagging this.

**Comment by eidam** [comment]
Thanks a lot!
